route.go

```diff
@@ -718,6 +718,25 @@ func (r *Route) GetHostTemplate() (string, error) {
 	return r.regexp.host.template, nil
 }
 
+// GetVarNames returns the names of all variables added by regexp matchers
+// These can be used to know which route variables should be passed into r.URL()
```

Comment:
Hi @eh-steve, can we add an example in the docs? I agree you already explain the use case briefly in the PR comment. Others who will go through the package documentation should understand its use case properly.

```go
func main() {

	r := mux.NewRouter()
	r.Host("{domain}").
		Path("/{group}/{item_id}").
		Queries("some_data1", "{some_data1}").
		Queries("some_data2", "{some_data2}").
		HandlerFunc(Func).
		Name("article")
	fmt.Println(r.Get("article").GetVarNames())
	http.ListenAndServe(":8000", r)

}
```` 

`GetVarNames` will print `[domain group item_id some_data1 some_data2] <nil>`

Reply:
Can we add an example in the `README.md` file and add test cases to validate the implementation.

Reply:
I've added some, not sure how useful they are, but lemme know?

Reply:
Added a review comment for the test case.